### DISCUSSION TITLE
Subquery in the FROM clause DOES not handle WHERE clause

### DISCUSSION THREAD

**Discussion by eaudet** [opendistro-for-elasticsearch/sql#949] [Q&A]
`SELECT * FROM ( SELECT subtract(2020, 2019) as sub, vintage as v FROM myindex ) o2 WHERE 2 = 1`

returns true all the time

**Comment by penghuo** [comment]
> `SELECT * FROM ( SELECT subtract(2020, 2019) as sub, vintage as v FROM myindex ) o2 WHERE 2 = 1`
> 
> returns true all the time

Hi. 
Have u enabled the new [engine](https://github.com/opendistro-for-elasticsearch/sql/blob/develop/docs/user/admin/settings.rst#opendistro-sql-engine-new-enabled)? 
In the new engine, we didn't support substract function, but we have "-" operator supported. then, the following query should work as expected, I tested it.
```
SELECT * FROM (SELECT 2020 - 2019 as sub FROM kibana_sample_data_logs ) o2 WHERE 2 = 1
```

**Reply by anirudha** [reply]
This is enabled by default now
